File: problems/121. Best Time to Buy and Sell Stock/solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if not prices: return 0
        minPrice = prices[0]
        maxProfit = 0
        
        for i in range(1, len(prices)):
            maxProfit = max(maxProfit, prices[i] - minPrice)
            minPrice = min(minPrice, prices[i])
        
        return maxProfit

File: problems/121. Best Time to Buy and Sell Stock/solution.cpp
class Solution
{
public:
    int maxProfit(vector<int> &prices)
    {
        if (prices.size() == 0)
            return 0;
        int minPrice = prices[0];
        int maxProfit = 0;
        for (int i = 1; i < prices.size(); i++)
        {
            minPrice = min(minPrice, prices[i]);
            maxProfit = max(maxProfit, prices[i] - minPrice);
        }

        return maxProfit;
    }
};
